File: lab-2.3/c#/Ironman.cs
namespace CharacterLib
{
    public class Ironman : IDescribable
    {
        private string name;
        private string suitModel;
        private int powerLevel;

        public Ironman()
        {
            name = "Unknown";
            suitModel = "Mark I";
            powerLevel = 100;
        }

        public Ironman(string name, string suitModel, int powerLevel)
        {
            this.name = name;
            this.suitModel = suitModel;
            this.powerLevel = powerLevel;
        }

        public Ironman(Ironman other)
        {
            name = other.name;
            suitModel = other.suitModel;
            powerLevel = other.powerLevel;
        }

        public string GetName()
        {
            return name;
        }

        public string GetSuitModel()
        {
            return suitModel;
        }

        public int GetPower()
        {
            return powerLevel;
        }

        public string Describe()
        {
            return $"Ironman: {name}, Suit: {suitModel}, Power: {powerLevel}";
        }
    }
}

File: lab-2.3/c++/Ironman.cpp
#include "Ironman.h"

Ironman::Ironman() : name("Unknown"), suitModel("Mark I"), powerLevel(100) {}

Ironman::Ironman(std::string n, std::string sm, int pl)
    : name(n), suitModel(sm), powerLevel(pl) {}

Ironman::Ironman(const Ironman& other)
    : name(other.name), suitModel(other.suitModel), powerLevel(other.powerLevel) {}

std::string Ironman::getName() const {
    return name;
}

std::string Ironman::getSuitModel() const {
    return suitModel;
}

int Ironman::getPower() const {
    return powerLevel;
}

std::string Ironman::describe() const {
    return "Ironman: " + name + ", Suit: " + suitModel + ", Power: " + std::to_string(powerLevel);
}

IDescribable* Ironman::clone() const {
    return new Ironman(*this);
}